src/linkml_arrays/dumpers/xarray_dumpers.py

```diff
@@ -0,0 +1,132 @@
+"""Class for dumping a LinkML model to netcdf like using xarray and DataTree."""
+
+from pathlib import Path
+
+import numpy as np
+from xarray.core.datatree import DataTree
+
+"""Class for dumping a LinkML model to an netcdf like file."""
+
+from pathlib import Path
+from typing import Union
+
+import xarray as xr
+import pandas as pd
+from linkml_runtime import SchemaView
+from linkml_runtime.dumpers.dumper_root import Dumper
+from linkml_runtime.utils.yamlutils import YAMLRoot
+from pydantic import BaseModel
+from linkml_runtime import SchemaView
+
+
+def _create_node(model, schemaview):
+    """Create datatree from temperature dataset"""
+    node_dict = {}
+    for k, v in vars(model).items():
+        if isinstance(v, str):
+            # parts of the dataset with key and value both being string, e.g. name, latitude_in_deg
+            try:
+                node_dict["attrs"][k] = v
+            except KeyError:
+                node_dict["attrs"] = {k: v}
+        elif isinstance(v, BaseModel):
+            if len(var_dict := vars(v)) == 1:
+                # If values are length 1 we are dealing with coords like date
+                v = v.values
+                try:
+                    node_dict["coords"][k] = v
+                except KeyError:
+                    node_dict["coords"] = {k: {"data": v, "dims": k}}
+            else:
+                for key, value in var_dict.items():
+                    if key == "values":
+                        if not isinstance(value[0], list):
+                            try:
+                                node_dict["coords"][k] = {"data": value, "dims": list(node_dict["coords"])[0]}
+                            except KeyError:
+                                node_dict["coords"] = {k: {"data": value, "dims": list(node_dict["coords"])[0]}}
+                        else:
+                            # Parse the temperature matrix
+                            element_type = type(v).__name__
+                            dimensions_expressions = schemaview.induced_slot(key, element_type).array.dimensions
+                            dims = [dim.alias for dim in dimensions_expressions]
+                            array = np.array(value)
+                            node_dict["dims"] = {dim: array.shape[i] for i, dim in enumerate(dims)}
+                            node_dict["data_vars"][k].update({"data": array, "dims": list(node_dict["dims"].keys())})
+                    else:
+                        if isinstance(value, str):
+                            # can't use timestamp here as it does not serialize, potentially add with 'data' dims as coord
+                            node_dict["coords"][k].update({"attrs": {key: value}})
+                        else:
+                            # conversion factor
+                            node_dict["data_vars"] = {k: {"attrs": {key: value}}}
```

Comment:
What if there is more than one? Here the attrs dict is set to a dictionary with just this kv pair. I think this should call `_create_node` recursively.